Unix/http/httpclientauth.c

```diff
@@ -211,13 +217,215 @@ typedef struct _Gss_Extensions
+
+  static Probable_Cause_Data AUTH_ERROR_CRED_FILE_PERMS = {
+              ERROR_ACCESS_DENIED,
+              WSMAN_CIMERROR_PROBABLE_CAUSE_AUTHENTICATION_FAILURE,
+              MI_T("credentials file permissions must be user-only") 
+       };
+
+
+    if (ntlm_user_file)
+    {
+        // Check the permissions on the directory. They need to be read-only owner
+        
+        {
+            struct stat buf = {0};
+            char *ntlm_cred_dir = strdup(ntlm_user_file);
+            char* p = strrchr(ntlm_cred_dir, '/');
+            if (p)
+            {
+                *p = '\0';
+            }
+
+            int rtn = stat(ntlm_cred_dir, &buf);
+            if (rtn < 0)
+            {
+                _g_gssClientState.probableCause = &AUTH_ERROR_CRED_FILE;
+                goto Err;
+               
+            }
+
+            if (!S_ISDIR(buf.st_mode))
+            {
+               // Not a directory? Unlikely, but complain and issue error
+                _g_gssClientState.probableCause = &AUTH_ERROR_CRED_FILE;
+                goto Err;
+            }
+
+            // Acceptable dir will be user only, no others
+            if (!(buf.st_mode & S_IRUSR) || ( buf.st_mode & (S_IRWXG|S_IRWXO)))
+            {
+                _g_gssClientState.probableCause = &AUTH_ERROR_CRED_DIR_PERMS;
+                goto Err;
+            }
+            free(ntlm_cred_dir);
+        }
+            
+        // Check the file for the same permissions
+        {
+            struct stat buf = {0};
+            int rtn = stat(ntlm_user_file, &buf);
+
+            if (rtn < 0)
+            {
+               // No directory? Complain and issue error
+                _g_gssClientState.probableCause = &AUTH_ERROR_CRED_DIR;
+                goto Err;
+            }
+
+            if (S_ISDIR(buf.st_mode) || S_ISLNK(buf.st_mode))
+            {
+               // Not a file? Unlikely, but complain and issue error
+                _g_gssClientState.probableCause = &AUTH_ERROR_CRED_FILE;
+                goto Err;
+            }
+
+            // Acceptable dir will be user only, no others
+            if (!(buf.st_mode & S_IRUSR) || ( buf.st_mode & (S_IRWXG|S_IRWXO)))
+            {
+                _g_gssClientState.probableCause = &AUTH_ERROR_CRED_DIR_PERMS;
+                goto Err;
+            }
+        }
+
+    }
+    else 
+    {
+        const static char OMI_DIR_NAME[] = "/.omi";
+        const static int  OMI_DIR_NAME_LEN = (sizeof(OMI_DIR_NAME)-1)/sizeof(char);  // the division is a noop, but present if we ever use wide chars
+        const static char NTLM_CRED_FILE_NAME[] = "/ntlmcred";
+        const static int  NTLM_CRED_FILE_NAME_LEN = (sizeof(NTLM_CRED_FILE_NAME)-1)/sizeof(char);
+
+
+        char *home_dir = getenv("HOME");
+        int  homedir_len = strlen(home_dir);
+        char *omi_dir = PAL_Malloc(homedir_len+OMI_DIR_NAME_LEN+1);
+        
+        {
+            char *pstr = omi_dir;
+            strcpy(pstr, home_dir);
```

Comment:
Hmm, this will core if `$HOME` isn't defined. Perhaps you should verify that it is defined and, if not, return an error?